src/permission/permission.h

```diff
@@ -0,0 +1,86 @@
+#ifndef SRC_PERMISSION_PERMISSION_H_
+#define SRC_PERMISSION_PERMISSION_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_options.h"
+#include "permission/child_process_permission.h"
+#include "permission/fs_permission.h"
+#include "permission/permission_node.h"
+#include "permission/worker_permission.h"
+#include "v8.h"
+
+#include <iostream>
+#include <map>
+
+namespace node {
+
+class Environment;
+
+namespace permission {
+
+#define THROW_IF_INSUFFICIENT_PERMISSIONS(env, perm_, resource_, ...)          \
```

Comment:
The problem is that doesn't always throw. It's a conditional that requires some permission context. I feel including it there would increase the complexity to read the permission code.

Reply:
Whatever happens here exactly, one thing we should keep in mind is to add a `#define THROW_IF_INSUFFICIENT_PERMISSIONS() do {} while(0)`  dummy in a backport of all usages of this macro so that there aren’t conflicts for future backports of just about anything that touches the C++ source.